docs/l10n.rst

```diff
@@ -9,17 +9,421 @@ Localization
+"group" comments to indicate strings required for activation. A group comment only ends when
+another group comment starts however, so you should either group your required strings at the
+bottom of a file, or also have a "not required" group comment. Here's an example:
+
+.. code-block:: fluent
+
+    ### File for example.html
+
+    ## Required
+    example_page_title = The Page Title
+    example_page_desc = This page is a test.
+
+    ## Not required
+    example_footer = This string isn't as important
+
+
+Any group comment (a comment that starts with "##") that starts with "Required" (case does not
+matter) will start a required strings block, and any other group comment will end it.
+
+Once you have your strings in your .ftl file you can place them in your template. We'll use the
+above .ftl file for a simple Jinja template example:
+
+.. code-block:: jinja
+
+    <!doctype html>
+    <html>
+    <head>
+        <title>{{ ftl('example-page-title') }}</title>
+    </head>
+    <body>
+        <h1>{{ ftl('example-page-title') }}</h1>
+        <p>{{ ftl('example-page-desc') }}</p>
+        <footer>
+            <p>{{ ftl('example-footer') }}</p>
+        </footer>
+    </body>
+    </html>
+
+.. _Fluent file syntax: https://projectfluent.org/fluent/guide/
+
+FTL String IDs
+~~~~~~~~~~~~~~
+
+Our convention for string ID creation is the following:
+
+1. String IDs should be all lower-case alphanumeric characters.
+2. Words should be separated with hyphens.
+3. IDs should be prefixed with the name of the template file (e.g. ``firefox-new-skyline`` for ``firefox-new-skyline.html``)
+4. If you need to create a new string for the same place on a page and to transition to it as it is translated, you can
+   add a version suffix to the string ID: e.g. ``firefox-new-skyline-main-page-title-v2``.
+4. The ID should be as descriptive as possible to make sense to the developer, but could be anything as long as it adheres
+   to the rules above.
+
+The ``ftl`` helper function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The ``ftl()`` function takes a string ID and returns the string in the current language,
+or simplified english if the string isn't translated. If you'd like to use a different
+string ID in the case that the primary one isn't translated you can specify that like this:
+
+.. code-block:: python
+
+    ftl('primary-string-id', fallback='fallback-string-id')
+
+When a fallback is specified it will be used only if the primary isn't translated in the current
+locale. English locales (e.g. en-US, en-GB) will never use the fallback and will print the simplified
+english version of the primary string if not overridden in the more specific locale.
+
+You can also pass in replacement variables into the ``ftl()`` function for use with `fluent variables`_.
+If you had a variable in your fluent file like this:
+
+.. code-block:: fluent
+
+    welcome = Welcome, { $user }!
+
```

Comment:
Should we make it explicit that you can't use `fallback` as a variable name? They're right next to each other, but maybe that's still good to call out?